### PR TITLE
feat(commerce/backend): add product and order endpoints

### LINKED ISSUE #620: Add routes for commerce backend

### PR DESCRIPTION
**Pull request by raducristianpopa**
### Context

- closes #620 

### Changes
This PR adds a couple of endpoints so we can start coupling the frontend to the backend.

### CODE REVIEW

File: packages/commerce/backend/src/product/controller.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      const { params } = req
+      if (!params.slug) {
+        this.logger.error(
+          'Attempting to fetch a product without the slug parameter provided.'
+        )
+        throw new BadRequest('Product slug was not provided.')
+      }
+
+      const product = await this.productService.getBySlug(params.slug)
+      if (!product) {
```
Comment (dragosp1011) [inline packages/commerce/backend/src/product/controller.ts:38]: throw the error form service method

File: packages/commerce/backend/src/product/controller.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ) {}
+
+  public async get(
+    req: Request<GetParams>,
+    res: TypedResponse<Product>,
+    next: NextFunction
+  ) {
+    try {
+      const { params } = req
+      if (!params.slug) {
```
[the remark this answers is not in the document]
Comment (raducristianpopa) [reply packages/commerce/backend/src/product/controller.ts:30]: That is my intention. Just keeping things simple at the moment and will do this incrementally.

File: packages/commerce/backend/tests/product/controller.test.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import { createContainer, type Cradle } from '@/container'
```
Comment (Tymmmy) [inline packages/commerce/backend/tests/product/controller.test.ts:1]: Tests, yaaaaaay

**Review by Tymmmy — DISMISSED** [review]

File: packages/commerce/backend/src/order/controller.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    next: NextFunction
+  ) {
+    try {
+      const { params } = req
+      if (!params.id) {
+        throw new BadRequest('Order ID was not provided.')
+      }
+
+      const order = await this.orderService.get(params.id)
+      if (!order) {
```
Comment (dragosp1011) [inline packages/commerce/backend/src/order/controller.ts:34]: move to service

File: packages/commerce/backend/migrations/20230824054747_add_products.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+const products = [
```
Comment (dragosp1011) [inline packages/commerce/backend/migrations/20230824054747_add_products.js:1]: it may be easier to manage if it will be extracted in a separate JSON file

**Review by dragosp1011 — APPROVED** [review]

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**